File: Concepts/LinkedLists/C/LinkedList.c
#include <stdio.h>
#include <conio.h>
#include <stdlib.h>

typedef struct LinkedList
{
    int data;
    struct LinkedList* next;
}Node;

void traversal(Node* head) {
    if (head == NULL)
        printf("List is empty.");
        
        printf("%d", head);
        if (head->next == NULL)
            return;
        head = head->next;
    }
    
    Node insertion(Node* newNode, int o) {
        if (newNode = NULL) {
            newNode->data = o;
            newNode->next = NULL;
            return *newNode;
        }
        
    }
    
    Node* deletion(Node* head, int o){
    if (head == NULL)
        printf("List is empty.");
    return head;
    
}

int main() {
    char op = '0';
    Node* head;
    int n = 0;
    while (op != '1') {
        system("cls");
        printf("1. Create a linked list\n");
        printf("2. Exit.\n");
        printf("Choose option: ");
        op = getch();
        if (op == '1')
            head = NULL;
        else if (op == '2')
            return 0;
    }
    
    while (op == 1 || op == 2 || op == 3)
	{
		printf("\n1. Traversal\n");
		printf("2. Insertion\n");
		printf("3. Deletion\n");
		printf("Press any key and Enter to exit\n");
		printf("\n Enter option: ");
        op = getch();
		switch (op)
		{
		case '1':
			traversal(head);
			break;
		case 2:
            printf("Enter new node: ");
            scanf("%d", &n);
			*head = insertion(head, n);
			break;
        case '3':
            printf("Enter index to delete: ");
            scanf("%d", &n);
			*head = deletion(head, n);
			break;
		default:
			printf("\nUnknown input\n");
			break;
		}
		getch();
		system("cls");

    }
    
    return 0;
}

File: Concepts/LinkedLists/Python/LinkedList.py
class Node:
    def __init__(self, val: int):
        self.item = val
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
        while True:
            print("\n----------------------------")
            print("1. Insert At Head")
            print("2. Insert At Tail")
            print("3. Delete At Head")
            print("4. Delete At Tail")
            print("5. Traverse")
            print("(Press any key to exit)")
            print("----------------------------")
            print("Enter an option: ")
            print("----------------------------\n")
            match (int(input())):
                case 1:
                    self.insert_at_head(int(input("Enter the value: ")))
                case 2:
                    self.insert_at_tail(int(input("Enter the value: ")))
                case 3:
                    self.delete_at_head()
                case 4:
                    self.delete_at_tail()
                case 5:
                    self.printList()
                case _:
                    break
    
    def insert_at_head(self, val: int):
        newNode = Node(val)
        newNode.next = self.head
        self.head = newNode
    
    def insert_at_tail(self, val: int):
        newNode = Node(val)
        if self.head == None:
            head = newNode
        currNode = self.head
        while currNode.next != None:
            currNode = currNode.next
        currNode.next = newNode

    def delete_at_tail(self):
        currNode = self.head
        while currNode.next.next != None:
            currNode = currNode.next
        currNode.next = None
    
    def delete_at_head(self):
        self.head = self.head.next
    
    def printList(self):
        if self.head == None:
            print("List is empty")
        currNode = self.head
        while currNode != None:
            print(currNode.item, " -> ", end="")
            currNode = currNode.next
        print("null")
        
ll = LinkedList()
